src/python/pants/help/help_printer.py

```diff
@@ -81,14 +73,91 @@ def print_hint() -> None:
             return 1
         return 0
 
+    def _print_alternatives(self, match: str, all_things: Iterable[str]) -> None:
+        did_you_mean = list(difflib.get_close_matches(match, all_things))
+
+        if did_you_mean:
+            formatted_matches = self._format_did_you_mean_matches(did_you_mean)
+            print(f"Did you mean {formatted_matches}?")
+
     def _print_title(self, title_text: str) -> None:
         title = self.maybe_green(f"{title_text}\n{'-' * len(title_text)}")
         print(f"\n{title}\n")
 
+    def _print_table(self, table: Dict[str, Optional[str]], indent: int = 0) -> None:
+        longest_key = max(len(key) for key, value in table.items() if value is not None)
+        for key, value in table.items():
+            if value is None:
+                continue
+            print(
+                self.maybe_cyan(f"{key:{longest_key}}:"),
+                self.maybe_magenta(
+                    f"\n{' ':{longest_key+2}}".join(
+                        hard_wrap(value, width=self._width - longest_key - 2)
+                    )
+                ),
+            )
+
+    def _get_thing_help_table(self) -> Dict[str, Callable[[str, bool], None]]:
+        def _help_table(
+            things: Iterable[str], help_printer: Callable[[str, bool], None]
+        ) -> Dict[str, Callable[[str, bool], None]]:
+            return dict(zip(things, cycle((help_printer,))))
+
+        top_level_help_items = _help_table(self._reserved_names, self._print_top_level_help)
+        return {
+            **top_level_help_items,
+            **_help_table(self._all_help_info.scope_to_help_info.keys(), self._print_options_help),
+            **_help_table(
+                self._all_help_info.name_to_target_type_info.keys(), self._print_target_help
+            ),
+            **_help_table(
+                self._all_help_info.name_to_api_type_info.keys(), self._print_api_type_help
+            ),
+            **_help_table(self._all_help_info.name_to_rule_info.keys(), self._print_rule_help),
+        }
```

Comment:
This table has the added benefit that we can calculate the `Did you mean...` suggestion using all possible thing values to give help on.